docs/api/agents.md

```diff
@@ -0,0 +1,439 @@
+# Agents
+
+> This page is incomplete, stay tuned.
+
+## Authenticate agent on AWS instance
+
+### Code samples
+
+```shell
+# Example request using curl
+curl -X POST http://coder-server:8080/api/v2/workspaceagents/aws-instance-identity \
+  -H 'Content-Type: application/json' \
+  -H 'Accept: application/json' \
+  -H 'Coder-Session-Token: API_KEY'
+```
+
+`POST /workspaceagents/aws-instance-identity`
+
+> Body parameter
+
+```json
+{
+  "document": "string",
+  "signature": "string"
+}
+```
+
+### Parameters
+
+| Name   | In   | Type                                                                             | Required | Description             |
+| ------ | ---- | -------------------------------------------------------------------------------- | -------- | ----------------------- |
+| `body` | body | [codersdk.AWSInstanceIdentityToken](schemas.md#codersdkawsinstanceidentitytoken) | true     | Instance identity token |
+
+### Example responses
+
+> 200 Response
+
+```json
+{
+  "session_token": "string"
+}
+```
+
+### Responses
+
+| Status | Meaning                                                 | Description | Schema                                                                                               |
+| ------ | ------------------------------------------------------- | ----------- | ---------------------------------------------------------------------------------------------------- |
+| 200    | [OK](https://tools.ietf.org/html/rfc7231#section-6.3.1) | OK          | [codersdk.WorkspaceAgentAuthenticateResponse](schemas.md#codersdkworkspaceagentauthenticateresponse) |
+
+To perform this operation, you must be authenticated by means of one of the following methods: **CoderSessionToken**.
```

Comment:
Idea: Should we link to the auth doc in these sentences?

Reply:
Fixed